src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -262,7 +262,11 @@ public void onSuccess(List<StepExecution> result) {
                     }
                     LOGGER.log(Level.FINE, "Will resume {0}", result);
                     for (StepExecution se : result) {
-                        se.onResume();
+                        try {
+                            se.onResume();
+                        } catch (Throwable x) {
+                            se.getContext().onFailure(x);
```

Comment:
Looking at this again, it might be better to use `se.stop` instead, although I'm not sure that it matters for any existing steps besides `node`. Compare to semi-related changes in https://github.com/jenkinsci/workflow-cps-plugin/pull/780 and https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/344.